main.tf

```diff
@@ -1,136 +1,179 @@
- *     },
- *     {
- *       size        = "1000"
- *       type        = ""     # Use AWS default.
- *       iops        = "0"    # Use AWS default.
- *       device_name = "/dev/xvdj"
- *     }
- *   ]
- * }
- *```
- */
-
-provider "aws" {}
-
-// If name_prefix exists, merge it into the cluster_name
-locals {
-  cluster_name = "${var.name_prefix != "" ? "${var.name_prefix}-${var.cluster_name}" : var.cluster_name}"
-
-  # NOTE: This is to workaround the divide by zero warning from Terraform.
-  num_extra_volumes = "${length(var.extra_volumes) > 0 ? length(var.extra_volumes) : 1}"
-
-  # NOTE: This is to make "lookup" happy. Otherwise, it will complain
-  # about the type cannot be derived if "var.extra_volumes" is not
-  # set.
-  extra_volumes = "${concat(var.extra_volumes, list(map("dummy", "dummy")))}"
+# :bangbang: THIS IS AN ALPHA MODULE DO NOT USE IN PRODUCTION :bangbang:
+# ======================================================================
+
+# AWS Windows Instance
+# ============
+# This is an module to creates a DC/OS Windows AWS Instances.
+
+# f `ami` variable is not set. This module uses the latest AWS Windows AMI.
+
+# Using you own AMI *FIXME*
+# -----------------
+# If you choose to use your own AMI please make sure the DC/OS related
+# prerequisites are met. Take a look at https://docs.mesosphere.com/1.11/installing/ent/custom/system-requirements/install-docker-RHEL/
+
+# EXAMPLE
+# -------
+
+# ```hcl
+# module "windows-agent" {
+#   source = "git::https://github.com/alekspv/terraform-aws-windows-instance.git?ref=features/windows-agent"
+#   num_winagent            = "2"
+#   admin_ips               = ["198.51.100.0/24"]
+#   vpc_id                  = "vpc-123456789"
+#   subnet_id               = "subnet-123456789"
+#   cluster_name            = "development"
+#   expiration              = "24h"
+#   owner                   = "John Dou"
+#   aws_key_name            = "${module.dcos.infrastructure.aws_key_name}"
+#   security_group_admin    = "${module.dcos.infrastructure.security_groups.admin}"
+#   security_group_internal = "${module.dcos.infrastructure.security_groups.internal}"
+#   bootstrap_private_ip    = "${module.dcos.infrastructure.bootstrap.private_ip}"
+#   bootstrap_public_ip     = "${module.dcos.infrastructure.bootstrap.public_ip}"
+#   bootstrap_os_user       = "${module.dcos.infrastructure.bootstrap.os_user}"
+#   ssh_private_key_file    = "~/.ssh/id_rsa"
+#   masters_private_ips     = "${module.dcos.infrastructure.masters.private_ips}"
+# }
+
+# ```
+
+
+data "aws_ami" "winAmi" {
+ owners                               = ["amazon"]
+  filter{
+    name                              = "name"
+    values                            = ["Windows_Server-1809-English-Core-ContainersLatest-2019.05.15"]
+  }
 }
 
-module "dcos-tested-oses" {
-  source  = "dcos-terraform/tested-oses/aws"
-  version = "~> 0.2.0"
-
-  providers = {
-    aws = "aws"
+resource "aws_security_group" "rdp" {
```

Comment:
we have a special module handing security groups. Those get forwarded by variable